PathOfBuildingCommunity/PathOfBuilding
# Add poeplanner.com tree import ability

Fixes #3767

### Description of the problem being solved:
Fix my frustration (and maybe someone else's) with:
![image](https://github.com/PathOfBuildingCommunity/PathOfBuilding/assets/4302241/eb11a648-90f3-4e30-b504-310627b96c24)
when importing the poe export tree supplied by poeplanner.com

### Description of the problem being solved:
Inability to import from poeplanner.com

## Note:
- poeplanner.com url also contains the bandit information. This is commented out. Is there a way to note if this is new build ?
- There are still debug prints in the code. This is so when someone finds something broken, I can check (more easily) what went wrong.
- Cluster nodes don't import properly, same as ggg urls.
- Attempting to decode the minified url (eg: https://poeplanner.com/b/E8I) fails with "We're sorry but PoE Planner doesn't work properly without JavaScript enabled. Please enable it to continue." returned.

### Steps taken to verify a working solution:
- Copy the poeplanner.com url, not the poe export url.
![image](https://github.com/PathOfBuildingCommunity/PathOfBuilding/assets/4302241/841c8f77-60fe-4712-bdfe-45ac9d407ceb)
- Import the tree using the PoB "Import Tree" button as per regular usage.
- Observe no error, and the tree is displayed, with Masteries if they were part of the import.

### Link to a build that showcases this PR:
These are some test links used.
- Scion, One node
https://www.poeplanner.com/AwAAAwAaAAAAAAEA1rcAAAAAAAAEAAD______________________wAAAgAAAAMA_____________zwAAAAAFAAfiwgAAAAAAAADAwAAAAAAAAAAAA==

- Scion, Four nodes
https://poeplanner.com/AwAAAwAaAAAAAAQAwyFl-daQ1rcAAAAAAAAEAAD______________________wAAAgAAAAMA_____________zwAAAAAFAAfiwgAAAAAAAADAwAAAAAAAAAAAA==

- Scion, Four nodes, one mastery, 3.21
https://poeplanner.com/AwAAAwAaAAAAAAQAwyFl-daQ1rcAAAAAAQBAR-DIBAAA______________________8AAAIAAAADAP____________88AAAAABQAH4sIAAAAAAAAAwMAAAAAAAAAAAA==

- Scion, Four nodes, one mastery, 3.17
https://poeplanner.com/AwAAAwAWAAAAAAQA1pDDIda3ZfkAAAAAAQBAR-DIBAAA______________________8AAAIAAAADAP____________88AAAAABQAH4sIAAAAAAAAAwMAAAAAAAAAAAA==

- Scion, Four nodes, one mastery, one ascendancy, 3.21
https://poeplanner.com/AwAAAwAaAAABAAQAwyFl-daQ1rcAAAEAreIBAEBH4MgEAAD______________________wAAAgAAAAMA_____________zwAAAAAFAAfiwgAAAAAAAADAwAAAAAAAAAAAA==

- Marauder, One node
https://poeplanner.com/AwAAAwAaAAEAAAEAjHsAAAAAAAAEAAD______________________wAAAgAAAAMA_____________zwAAAAAFAAfiwgAAAAAAAADAwAAAAAAAAAAAA==

- Marauder, Jugg, One node
https://poeplanner.com/AwAAAwAaAAECAAEAjHsAAAIAEOqQIQAABAAA______________________8AAAIAAAADAP____________88AAAAABQAH4sIAAAAAAAAAwMAAAAAAAAAAAA==

- Marauder, Jugg, One Cluster
https://poeplanner.com/AwAAAwAaAAEDAA8A2MZoeW_zY1g_X9VJSPYEweg2WMRuqbsJ_D2vWGCGBwB2AHoAfAB-AIAAeAB7AAgA-X3DBqOm8yZ-uYwXZ88ExgAABAACAZYDVGhyZWFkIG9mIEhvcGUAAP8AAAD__wNzjAEDAHaBAQEAbIEB9v8AAAAAAAA-AwAA_wAAAAP8AAEBAEMBAQEAaAEBAQAAAmudAQIAf50BCAABdp0BCgAAAP______________________AgC7CQEAb9MAAAIAAAADAP____________88AAAAABQAH4sIAAAAAAAAAwMAAAAAAAAAAAA==

- Marauder, Jugg, One Cluster, One Mastery
https://poeplanner.com/AwAAAwAaAAEDABIA2MZm6Pw_aHlv82NYP1-xCNVJSPYEweg2WMRuqbsJ_D2vWGCGBwB2AHoAfAB-AIAAeAB7AAgA-X3DBqOm8yZ-uYwXZ88ExgEAlYn94AQAAgGWA1RocmVhZCBvZiBIb3BlAAD_AAAA__8Dc4wBAwB2gQEBAGyBAfb_AAAAAAAAPgMAAP8AAAAD_AABAQBDAQEBAGgBAQEAAAJrnQECAH-dAQgAAXadAQoAAAD______________________wIAuwkBAG_TAAACAAAAAwD_____________PAAAAAAUAB-LCAAAAAAAAAMDAAAAAAAAAAAA==

- Oni Goroshi Tectonic Slam, from https://www.poebuilds.net/builds/marauder/oni-goroshi-tectonic-slam
https://poeplanner.com/AwAAAwAaAAEDAGkA-zGsdmAhGOpfnSooCyd-1zoC2Ma1ZHUUD0nZhO273Tk8BTwOfNktBZ_jzpGqGxj6p74wOG_zY1gHwbtTjNxxFFATMBKgZ2YFMlqNrWVoPj1SOj9fYdkXYwiXoEDP5ZEY6_gaWtVJ439iNp5E53bhMUdQUoXHzLHkSPbRhlM4b-52h5-82umvW0363fNO7-96UmFY_m_T8XRR5Og2WMThEnrufUrbGK8i13e3RnmXuwl9khNeB4IKzzL3_D0gFFMuD5Nghu_hfs_tdE-LtoPtiA7vBwB2AHoAfAB-AIAAeAB7AAgA-X3DBqOm8yZ-uYwXZ88ExgsASWjqxn_JAicrEBq60PZSH7CVdF7bvmv9x-c1bx-Eqwtt6Hfg-lAFsYD63m8EAAIBlgNUaHJlYWQgb2YgSG9wZQAA_wAAAP__A3OMAQMAdoEBAQBsgQH2_wAAAAAAAD4DAAD_AAAAA_wAAQEAQwEBAQBoAQEBAAACa50BAgB_nQEIAAF2nQEKAAAA______________________8CALsJAQBv0wAAAgAAAAMA_____________zwAAAAAFAAfiwgAAAAAAAADAwAAAAAAAAAAAA==
